backend/app/src/main/java/com/moh/phlat/backend/esb/boundary/PlrEsbBoundary.java

```diff
@@ -0,0 +1,209 @@
+import org.springframework.boot.ssl.NoSuchSslBundleException;
+import org.springframework.boot.ssl.SslBundle;
+import org.springframework.boot.ssl.SslBundles;
+import org.springframework.http.MediaType;
+import org.springframework.http.client.reactive.JdkClientHttpConnector;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+
+import javax.net.ssl.SSLContext;
+import java.net.http.HttpClient;
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+public class PlrEsbBoundary {
+	
+	private static final Logger logger = LoggerFactory.getLogger(PlrEsbBoundary.class);
+	private WebClient webClient;
+
+	private PlrToken token;
+	
+	private SSLContext sslContext;
+
+	@Autowired
+    public void initSslContext(SslBundles sslBundles) throws NoSuchSslBundleException {
+		SslBundle sslBundle = sslBundles.getBundle("client");
+        sslContext = sslBundle.createSslContext();
+    }
+	
+	@Value("${plr.boundary.host}")
+	private String plrBoundaryHost;
+
+	@Value("${plr.boundary.keycloak.provider-url}")
+	private String keyCloakUrl;
+	@Value("${plr.boundary.keycloak.client-id}")
+	private String plrBoundaryClientId;
+	@Value("${plr.boundary.keycloak.client-secret}")
+	private String plrBoundaryClientSecret;
+	
+	private final String plrBoundaryEndpoint = "/HSA-web/rs/passthrough/maintain";
+	
+	@PostConstruct
+	public void initPlrEsbBoundary() {
+		try {
+			HttpClient httpClient = HttpClient.newBuilder()
+					// *TEMPORARY SSL DISABLER*
+					.sslContext(sslContext)
+					.build();
+			
+			webClient = WebClient.builder()
+					.clientConnector(new JdkClientHttpConnector(httpClient))
+					.build();
+			
+		} catch (Exception ex) {
+			logger.error("Could not build PlrEsbBoundary Connection: ", ex);
+		}
+	}
+	
+	public boolean isReadyToConnect() {
+		return webClient != null;
+	}
+	
+	public List<PlrResponse> loadPlrViaEsb(Control control, ProcessData processData) {
+		List<PlrResponse> plrResponses = new ArrayList();
+		token = getPlrKeyCloakDetails();
+		if (StringUtils.hasText(token.getAccessToken()) && !StringUtils.hasText(token.getError())) {
+			if (control.getLoadTypeFacility()) {
+				MaintainFacilityResponse facilityResponse = createFacility(control, processData);
+				plrResponses.add(facilityResponse);
+			}
+			if (control.getLoadTypeHds()) {
+				//MaintainHdsResponse hdsResponse = createHdsProvider(control, processData);
+				//plrResponses.add(hdsResponse);
+			}
+			if (control.getLoadTypeOFRelationship()) {
+				//MaintainOFResponse ofResponse = createOFRelationship(control, processData);
+				//plrResponses.add(ofResponse);
+			}
+		} 
+		return plrResponses;
+	}
+	
+	private MaintainFacilityResponse createFacility(Control control, ProcessData processData) {
+		MaintainFacilityRequest maintainFacilityRequest = new MaintainFacilityRequest(control);
```

Comment:
if we are creating an instance of concrete class, then what is the use of Interface - PlrRequest?

Reply:
Unlike PlrResponse, you're probably right about this interface; it's not as useful and there aren't as many common elements. Again though, I think we can save this for later cleanup, and at the moment it at least exposes a shared date-format for the JSON object mapper